### PR TITLE
Support datafusion as a metadata source

### PR DESCRIPTION
**Pull request by jhoekx**

### CODE REVIEW

File: docs/source/datafusion.asciidoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+The column names returned by the `list_query` are used as-is to populate tags as present in `tag_columns` and known metadata fields.
+Columns with unknown names define extra metadata fields.
+
+Kukur configures DataFusion to be case sensitive.
+
+The following example defines a DataFusion source that connects to 3 tables:
+
+- A CSV files containing a list of all time series and their units.
```
Comment (lucasKoeb) [inline docs/source/datafusion.asciidoc:33]:
```suggestion
- A CSV file containing a list of all time series and their units.
```

File: docs/source/datafusion.asciidoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+type = "datafusion"
+list_query = "" # required: SQL query
+tag_columns = ["series name"]
+field_columns = ["value"]
+metadata_value_mapping = "" # optional: convert metadata values
+
+[[source.datafusion.table]]
+type = "csv|delta|json|parquet"
+name = "" # name of the table in SQL queries
+location = "" # path to a file or URI of a Delta Tabble
```
Comment (lucasKoeb) [inline docs/source/datafusion.asciidoc:23]:
```suggestion
location = "" # path to a file or URI of a Delta Table
```

**Review by lucasKoeb — APPROVED** [review]

**Comment by smeganck** [comment]
@lucasKoeb

**Comment by lucasKoeb** [comment]
Validated by configuring a datafusion source on timeseer and listing series.

**Pull request merged**